dpnp/dpnp_iface_linearalgebra.py

```diff
@@ -567,10 +612,19 @@ def tensordot(a, b, axes=2):
 
     dpnp.check_supported_arrays_type(a, b, scalar_type=True)
 
-    if dpnp.isscalar(a):
-        a = dpnp.array(a, sycl_queue=b.sycl_queue, usm_type=b.usm_type)
-    elif dpnp.isscalar(b):
-        b = dpnp.array(b, sycl_queue=a.sycl_queue, usm_type=a.usm_type)
+    if dpnp.isscalar(a) or dpnp.isscalar(b):
+        if axes > 0:
```

Comment:
@antonwolfy I believe the correct condition here is `if axes!=0` but I did not used it since  `NumPy` works with negative axes for such a case which seems to be a bug. Should we keep it as it is to replicate `NumPy` or modify it?
```
>>> numpy.tensordot(numpy.arange(5), 2, axes=-5)
array([0, 2, 4, 6, 8])
```

Reply:
do you mean something like cupy has here?
```python
    if a_ndim == 0 or b_ndim == 0:
        if axes != 0 and axes != ((), ()):
            raise ValueError('An input is zero-dim while axes has dimensions')
        return cupy.multiply(a, b)
```
I tend to agree with that. Moreover python array API [states](https://data-apis.org/array-api/2022.12/API_specification/generated/array_api.tensordot.html#array_api.tensordot) for `axes`:
>  ... The size of each corresponding axis (dimension) must match. **Must be nonnegative.**